pkg/providers/nutanix/template.go

```diff
@@ -239,6 +239,28 @@ func buildTemplateMapCP(
 		values["awsIamAuth"] = true
 	}
 
+	if clusterSpec.Cluster.Spec.ProxyConfiguration != nil {
+		values["proxyConfig"] = true
+		capacity := len(clusterSpec.Cluster.Spec.ClusterNetwork.Pods.CidrBlocks) +
+			len(clusterSpec.Cluster.Spec.ClusterNetwork.Services.CidrBlocks) +
+			len(clusterSpec.Cluster.Spec.ProxyConfiguration.NoProxy) + 4
+		noProxyList := make([]string, 0, capacity)
+		noProxyList = append(noProxyList, clusterSpec.Cluster.Spec.ClusterNetwork.Pods.CidrBlocks...)
+		noProxyList = append(noProxyList, clusterSpec.Cluster.Spec.ClusterNetwork.Services.CidrBlocks...)
+		noProxyList = append(noProxyList, clusterSpec.Cluster.Spec.ProxyConfiguration.NoProxy...)
+
+		// Add no-proxy defaults
+		noProxyList = append(noProxyList, clusterapi.NoProxyDefaults()...)
+		noProxyList = append(noProxyList,
+			datacenterSpec.Endpoint,
+			clusterSpec.Cluster.Spec.ControlPlaneConfiguration.Endpoint.Host,
+		)
+
+		values["httpProxy"] = clusterSpec.Cluster.Spec.ProxyConfiguration.HttpProxy
+		values["httpsProxy"] = clusterSpec.Cluster.Spec.ProxyConfiguration.HttpsProxy
+		values["noProxy"] = noProxyList
```

Comment:
For control plane and worker node, since we use the same noProxy list, can we get it from a common place by creating a function like [this](https://github.com/aws/eks-anywhere/blob/main/pkg/providers/tinkerbell/template.go#L641-L659)?

Reply:
Sure; fixed! PTAL!